src/components/Slide/Slide.js

```diff
@@ -0,0 +1,299 @@
+
+const GUTTER = 24;
+
+// Translate the node so he can't be seen on the screen.
+// Later, we gonna translate back the node to his original location
+// with `translate3d(0, 0, 0)`.`
+function getTranslateValue(props, node) {
+  const { direction } = props;
+  const rect = node.getBoundingClientRect();
+
+  let transform;
+
+  if (node.fakeTransform) {
+    transform = node.fakeTransform;
+  } else {
+    const computedStyle = ownerWindow(node).getComputedStyle(node);
+    transform =
+      computedStyle.getPropertyValue('-webkit-transform') ||
+      computedStyle.getPropertyValue('transform');
+  }
+
+  let offsetX = 0;
+  let offsetY = 0;
+
+  if (transform && transform !== 'none' && typeof transform === 'string') {
+    const transformValues = transform
+      .split('(')[1]
+      .split(')')[0]
+      .split(',');
+    offsetX = parseInt(transformValues[4], 10);
+    offsetY = parseInt(transformValues[5], 10);
+  }
+
+  if (direction === 'left') {
+    return `translateX(100vw) translateX(-${rect.left - offsetX}px)`;
+  }
+
+  if (direction === 'right') {
+    return `translateX(-${rect.left + rect.width + GUTTER - offsetX}px)`;
+  }
+
+  if (direction === 'up') {
+    return `translateY(100vh) translateY(-${rect.top - offsetY}px)`;
+  }
+
+  // direction === 'down'
+  return `translateY(-${rect.top + rect.height + GUTTER - offsetY}px)`;
+}
+
+export function setTranslateValue(props, node) {
+  const transform = getTranslateValue(props, node);
+
+  if (transform) {
+    node.style.webkitTransform = transform;
+    node.style.transform = transform;
+  }
+}
+
+/**
+ * The Slide transition is used by the [Snackbar](/demos/snackbars) component.
+ * It uses [react-transition-group](https://github.com/reactjs/react-transition-group) internally.
+ */
+class Slide extends React.Component {
+  componentDidMount() {
+    // state.mounted handle SSR, once the component is mounted, we need
+    // to properly hide it.
+    if (!this.props.in) {
+      // We need to set initial translate values of transition element
+      // otherwise component will be shown when in=false.
+      this.updatePosition();
+    }
+
+    this.mounted = true;
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.direction !== this.props.direction && !this.props.in) {
+      // We need to update the position of the drawer when the direction change and
+      // when it's hidden.
+      this.updatePosition();
+    }
+  }
+
+  componentWillUnmount() {
+    this.handleResize.clear();
+  }
+
+  handleResize = debounce(() => {
+    // Skip configuration where the position is screen size invariant.
+    if (
+      this.props.in ||
+      this.props.direction === 'down' ||
+      this.props.direction === 'right'
+    ) {
+      return;
+    }
+
+    // eslint-disable-next-line react/no-find-dom-node
+    const node = ReactDOM.findDOMNode(this.transition);
+    if (node) {
+      setTranslateValue(this.props, node);
+    }
+  }, 166); // Corresponds to 10 frames at 60 Hz.
```

Comment:
How about using [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) instead of `debounce`? It's a more elegant/modern way of achieving this and React uses it internally so it should already be polyfilled.

Otherwise, I would prefer using Lodash's [`debounce`](https://lodash.com/docs/4.17.10#debounce) to avoid adding another dependency.

Reply:
I don't see how `requestAnimationFrame` can be used in that case. Please let me know if you do.

Reply:
The `debounce` in that case is used to that the function is called at most once in a frame. Since `resize` event can be thrown many times in a single frame, calling `requestAnimationFrame` on each `resize` event occurrence might result in multiple calls of the function in a single frame.

Reply:
`requestAnimationFrame` returns an id which can be used to cancel the request with `cancelAnimationFrame`. So before making a new request, you can cancel any pending ones and thus avoid the multiple calls. [Here's an example use case on the website](https://github.com/sumup/website/pull/550/files#diff-a74f70dedc7bc1ddcf3422c1fd2f86e6R40).